virttest/qemu_vm.py

```diff
@@ -894,19 +894,26 @@ def set_dev_params(dev, dev_params,
 
         def add_memorys(devices, params):
             """
-            Add memory controler by params
+            Add memory controller by params
 
             :param devices: VM devices container
             """
             options, devs = [], []
             params.setdefault("mem", "512")
             mem_params = params.object_params("mem")
-            mem_params.setdefault("automem", "no")
+            mem_params.setdefault("automem", "yes")
             automem = mem_params["automem"] == "yes"
             if automem:
                 normalize_data_size = utils_misc.normalize_data_size
                 mem_size_m = "%sM" % mem_params["mem"]
                 mem_size_m = float(normalize_data_size(mem_size_m))
+                if mem_params.get("guest_max_mem"):
```

Comment:
I think it isn't associated with `automem`. What do you expect if `mem = 8G`, `automem = no` and `guest_max_mem = 4G`?